tools: rolled up lint-md

refack opened the pull request:
replace `remark-cli` and `remark-preset-lint-node` with a ["rolled-up"](https://rollupjs.org/guide/en) single file.

* Should solve all build & `Makefile` issue
* very size economic (17KB instead of 20MB ~1300 files)
* includes all recipes for re rolling & updating

/CC @wooorm @nodejs/build
H/T: @rollup

<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [X] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] documentation is changed or added
- [X] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

refack commented:
https://ci.nodejs.org/job/node-test-commit-lite/709/

vsemozhetbyt commented on tools/node-lint-md-cli-rollup/.gitignore:
```diff
@@ -0,0 +1,2 @@
+/node_modules/
+/dist/
```
Nit: missing line break at EOF.

richardlau commented on tools/remark-cli/LICENSE:
```diff
@@ -1,22 +0,0 @@
-(The MIT License)
```
Please could you update https://github.com/nodejs/node/blob/e86b319e3f9caa60502b0b0235c34ef0d7250c23/tools/license-builder.sh#L88?

gibfahn commented:
This sounds like a great solution to the problem. If it works we should probably consider doing the same for eslint.

joyeecheung commented on tools/node-lint-md-cli-rollup/package.json:
```diff
@@ -0,0 +1,23 @@
+  },
+  "scripts": {
+    "build": "rollup -c",
+    "build-node": "npm run build && cp dist/* .."
```
The `lint-md-build` is removed, can you add it back with a command that runs this?

joyeecheung commented:
Very nice work. Thank you for working on this!

BridgeAR commented:
This is marked as WIP @refack is there anything else to do to get this landed?

refack commented:
> refack is there anything else to do to get this landed (besides open comments)?

I was hoping for some feedback on how we should track the license, and document this. Also i'm not sure how to test the new script...

joyeecheung commented:
@refack I think writing a bad .md in the root directory then running `make lint-md` expecting a non-zero exit code and some output complaining about it should be enough. Can be part of the doctool test suite. If we don't want to always run it we can just tweak the makefile and/or vcbuild.bat, it should be fine as long as it's run in the linter CI.

BridgeAR commented:
Ping @refack

BridgeAR commented:
I guess it is a good idea to do this but I am going to close this due to no response. If anyone feels like taking a stab at this, that would be great.

@refack please reopen in case you want to continue working on this.

refack commented:
Rebased, updated, restructured commits (each are independently stable)
CI: https://ci.nodejs.org/job/node-test-pull-request/17067/

PTAL @jasnell @devsnek @joyeecheung @richardlau @nodejs/documentation 

/CC @wooorm

BridgeAR commented on Makefile:
```diff
@@ -1055,72 +1060,47 @@ bench-addons-clean:
-.PHONY: lint-md
-ifneq ("","$(wildcard tools/remark-cli/node_modules/)")
+lint-md-build: 
+	$(warning "Deprecated no-op target 'lint-md-build'")
```
Why should this not trigger the rollup?

refack replied:
It has different semantics, before it was necessary to run `lint-md-build` for `lint-md` to work. Now `lint-md-rollup` is more like `update-eslint.sh`. It needs to be run once in a while, and the results should be checked in.
Also It is currently run in CI for every run, and that is unnecessary.

refack commented:
> Why should this not trigger the rollup?

Also `lint-md-rollup` uses `npm up`, and there is some spesific workaround code in the rollup config that could break:
https://github.com/nodejs/node/blob/67b6082e119b85551e5dcf4243b09f08f50f42ce/tools/node-lint-md-cli-rollup/rollup.config.js#L28-L36

refack commented:
https://ci.nodejs.org/job/node-test-pull-request/17071/

refack commented:
Landed in
d3442f261ac60eafc948e681a23dc868e0b6d2da
2120dac8abc6d089903391735b00f12878fdf8ce
7fcc178be1766fcf6869152dd233b4184874ef8a
fe6c74c2a5c092b80b4ff988c6ca7883393f6b91
e4dd213516afdff64063682e0d0d46c58499a9f3
